### PR TITLE
feat: enhance analysis options with od model dump and better vis

### PR DESCRIPTION
**Pull request by pawel-kmiecik**
This PR adds new capabilities for drawing bboxes for each layout (extracted, inferred, ocr and final) + OD model output dump as a json file for better analysis.

### CODE REVIEW

**Review by christinestraub — CHANGES_REQUESTED** [review]
Drawing bboxes for OCR layout doesn't appear to be working.

PDF: [2023_SustainabilityReport_33.pdf](https://github.com/user-attachments/files/15917737/2023_SustainabilityReport_33.pdf)

```
elements = partition_pdf(
    filename="2023_SustainabilityReport_33.pdf",
    strategy=strategy,
    analysis=True,
)
```
**Results:**

page1_layout_ocr

page1_layout_od_model

page1_layout_pdfminer

page1_layout_final

**Comment by pawel-kmiecik** [comment]
> Drawing bboxes for OCR layout doesn't appear to be working.
> 
> PDF: [2023_SustainabilityReport_33.pdf](https://github.com/user-attachments/files/15917737/2023_SustainabilityReport_33.pdf)
> 
> ```
> elements = partition_pdf(
>     filename="2023_SustainabilityReport_33.pdf",
>     strategy=strategy,
>     analysis=True,
> )
> ```
> 
> **Results:**
> 
> page1_layout_ocr
> 
> page1_layout_od_model
> 
> page1_layout_pdfminer
> 
> page1_layout_final

This should be fixed now:
page1_layout_ocr

**Review by christinestraub — APPROVED** [review]
LGTM!

File: unstructured/partition/pdf_image/analysis/layout_dump.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    for page in layout.pages:
+        size = [page.image_metadata.get("width"), page.image_metadata.get("height")]
+        elements = []
+        for element in page.elements:
+            bbox = element.bbox
+            elements.append(
+                {
+                    "bbox": [bbox.x1, bbox.y1, bbox.x2, bbox.y2],
+                    "type": element.type,
```
Comment (mariannaparzych) [inline unstructured/partition/pdf_image/analysis/layout_dump.py:29]: those "types" I found in dumps are not compatible with `YOLO_SG_LABEL_MAP`, which I need to evaluate OD model,
I also need list of `YOLO_SG_LABEL_MAP` in json (as "categories" eg.) as I should not import it from _unstructured_inference_prop/object_detection/yolox_super_gradients.py_ in evaluate

Comment (pawel-kmiecik) [reply unstructured/partition/pdf_image/analysis/layout_dump.py:31]: Added in the last commit.

**Pull request merged**